src/domain/gift-cards/overview.tsx

```diff
@@ -0,0 +1,124 @@
+import { RouteComponentProps } from "@reach/router"
+import { navigate } from "gatsby"
+import { useAdminGiftCards, useAdminStore } from "medusa-react"
+import qs from "query-string"
+import React, { useContext, useEffect, useMemo, useState } from "react"
+import PageDescription from "../../components/atoms/page-description"
+import PlusIcon from "../../components/fundamentals/icons/plus-icon"
+import BannerCard from "../../components/molecules/banner-card"
+import BodyCard from "../../components/organisms/body-card"
+import DeletePrompt from "../../components/organisms/delete-prompt"
+import GiftCardBanner from "../../components/organisms/gift-card-banner"
+import GiftCardTable from "../../components/templates/gift-card-table"
+import { InterfaceContext } from "../../context/interface"
+import NewGiftCard from "./new"
+
+type OverviewProps = {
+  giftCard: any
+  onDelete: () => void
+  onUpdate: () => void
+} & RouteComponentProps
+
+const Overview: React.FC<OverviewProps> = ({
+  giftCard,
+  onDelete,
+  onUpdate,
+}) => {
+  const { store } = useAdminStore()
+  const { gift_cards: giftCards } = useAdminGiftCards()
+  const [showCreate, setShowCreate] = useState(false)
+  const [showDelete, setShowDelete] = useState(false)
+
+  const searchQuery = (q: string) => {
+    const baseUrl = qs.parseUrl(window.location.href).url
+
+    const search = {
+      fields: "id,title,thumbnail",
+      expand: "variants,variants.prices,collection",
+      q,
+      offset: 0,
+      limit: 20,
+    }
+
+    const prepared = qs.stringify(search, {
+      skipNull: true,
+      skipEmptyString: true,
+    })
+
+    window.history.replaceState(baseUrl, "", `?${prepared}`)
+    // TODO: Refresh
+  }
+
+  const actionables = [
+    {
+      label: "Custom Gift Card",
+      onClick: () => console.log("create custom gift card"), // TODO: open modal
```

Comment:
Is this part of another PR? You have a screenshot on your PR description which says "Create Gift Card" but I'm not sure if there is another way to make the modal appear other than this action :sweat_smile: . For now, I just turned the `showCreate` boolean to true to see the create gift card modal :smile:

Reply:
Well spotted @zakariaelas ! This is a separate ticket. Will make sure to create it 👍

Reply:
The button to show create new Gift Card is found on line 92, and it only shows if you do not have any gift cards in your store. That's properly why you couldn't open it 😄